### PR TITLE
HPCC-16462 Function with non-unique parameter names should give error

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: ecl/hql/hqlgram2.cpp [hunk 1]
```diff
@@ -6713,6 +6713,12 @@ bool HqlGram::checkParameters(IHqlExpression* func, HqlExprArray& actuals, const
 void HqlGram::addActiveParameterOwn(const attribute & errpos, IHqlExpression * param, IHqlExpression * defaultValue)
 {
     ActiveScopeInfo & activeScope = defineScopes.tos();
+    ForEachItemIn(idx, activeScope.activeParameters)
```
Comment (ghalliday) [inline ecl/hql/hqlgram2.cpp]: I suspect this may cause problems with nested functions.

**Comment by ghalliday** [comment]
@richardkchapman this needs a test case (in ecl/regress) that fails, and also an example with nested definitions which should be allowed to have duplicate names.

**Comment by richardkchapman** [comment]
@ghalliday Testcase added

File: ecl/hql/hqlgram2.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hql/hqlgram2.cpp]: I was thinking of ActiveScopeInfo::firstSideEffect - this code should be fine.

**Pull request merged**